### PR TITLE
Globalstore interface

### PR DESCRIPTION
**Pull request by biscon**
-Changed GlobalStore to be an interface, allowing us to mock it and change the implementation without changing code in the clients
- Made GlobalStoreImpl internal and made a factory function createGlobalStore
- Changed getSubscriberCount to a val property called subscriberCount
- Updated and ran tests

### CODE REVIEW

File: arch/src/main/java/dk/ufst/arch/Store.kt [hunk 1]
```diff
@@ -8,21 +8,27 @@ interface LocalStore<Value, Action> {
...
 
+interface GlobalStore<Value, Action, Environment> {
+    fun sendAction(action: Action)
+    fun subscribe(subscriber: (Value) -> Unit)
+    fun desubscribe(subscriber: (Value) -> Unit)
+    val subscriberCount: Int
+    val value: Value
+}
+
 @Suppress("unused", "MemberVisibilityCanBePrivate")
```
Comment (patrickusiewicz) [inline arch/src/main/java/dk/ufst/arch/Store.kt:19]: Not super important, but it looks like we can remove `MemberVisibilityCanBePrivate`?

File: arch/src/main/java/dk/ufst/arch/Store.kt [hunk 2]
```diff
@@ -64,6 +71,23 @@ open class GlobalStore<Value, Action, Environment>(
     }
 
     fun destroy() {}
```
Comment (patrickusiewicz) [inline arch/src/main/java/dk/ufst/arch/Store.kt:73]: Is `destroy` still needed?

**Review by patrickusiewicz — APPROVED** [review]
Nice interface addition! This will make testing including `GlobalStore` easier!

File: arch/src/main/java/dk/ufst/arch/Store.kt (hunk 2 above)
Comment (biscon) [reply arch/src/main/java/dk/ufst/arch/Store.kt:73]: no I just didn't want to fix stuff unrelated to the subject of the PR, but guess I did anyway by changing getSubscriberCount haha so why not :)

**Pull request merged**